### PR TITLE
Improve JWT verification error reporting

### PR DESCRIPTION
**Pull request by qiwzhang**
* Pass the detail error in the response for easy debugging.
* Add a tool verify_token to test token verification in a standalone environment.

### CODE REVIEW

File: src/nginx/t/auth_pkey.t [hunk 1]
```diff
@@ -113,7 +113,7 @@ foreach my $pkey($pkey_jwk, $pkey_x509) {
   like($response, qr/Content-Type: application\/json/i,
        'Missing creds returned application/json body.');
   like($response, qr/JWT validation failed: Missing or invalid credentials/i,
-       "Error body contains 'Missing or invalid credentials'.");
+                  "Error body contains 'Missing or invalid credentials'.");
```
Comment (kh-chang) [inline src/nginx/t/auth_pkey.t]: nit: indentation for `like` is not consistent.

File: src/tools/verify_token.cc [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+}  //  namespace
+
+void print_usage() {
+  std::cerr << "Invalid argument.\n"
+               "Usage: verify_token token_file public_key_file\n";
+}
+
+std::string read_file(const char *file) {
+  std::cerr << "==Reading file: " << file << std::endl;
```
Comment (kh-chang) [inline src/tools/verify_token.cc]: maybe std:cout? and also for `Token` and `Jwks` if you decide to change it

File: src/tools/verify_token.cc [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  buf[buf_len] = 0;
+  return std::string(buf, buf_len);
+}
+
+int main(int argc, char **argv) {
+  if (argc <= 2) {
+    print_usage();
+    return 1;
+  }
+  //  ::google::api_manager::auth::esp_init_grpc_log();
```
Comment (kh-chang) [inline src/tools/verify_token.cc]: del

File: src/api_manager/auth/lib/auth_jwt_validator.cc [hunk 4]
```diff
@@ -304,12 +304,12 @@ void JwtValidatorImpl::UpdateAudience(grpc_json *json) {
...
 }
 
-grpc_jwt_verifier_status JwtValidatorImpl::ParseImpl() {
+Status JwtValidatorImpl::ParseImpl() {
   // ====================
   // Basic check.
   // ====================
   if (jwt == nullptr || jwt_len <= 0) {
-    return GRPC_JWT_VERIFIER_BAD_FORMAT;
+    return ToStatus("Bad JWT format: empty token");
```
Comment (JLXIA) [inline src/api_manager/auth/lib/auth_jwt_validator.cc]: what is the error code by using this ToStatus?

File: src/api_manager/auth/lib/auth_jwt_validator_test.cc [hunk 5]
```diff
@@ -391,44 +391,47 @@ class JwtValidatorTest : public ::testing::Test {
...
   std::unique_ptr<JwtValidator> validator =
       JwtValidator::Create(token, strlen(token));
   Status status = validator->Parse(&user_info);
-  ASSERT_TRUE(status.ok());
-  ASSERT_EQ(status.message(), "");
-  ASSERT_EQ(kUserId, user_info.id);
-  ASSERT_TRUE(user_info.audiences.end() != user_info.audiences.find(kAudience));
-  ASSERT_EQ(1U, user_info.audiences.size());
-  ASSERT_EQ(kUserId, user_info.issuer);
+  EXPECT_TRUE(status.ok());
```
Comment (JLXIA) [inline src/api_manager/auth/lib/auth_jwt_validator_test.cc]: ->EXPECT_OK(status)?

File: src/tools/BUILD [hunk 6]
```diff
@@ -56,7 +56,17 @@ cc_binary(
     ],
     visibility = ["//visibility:public"],
     deps = [
-        "//external:api_manager_auth_lib",
+        "//src/api_manager/auth/lib",
+    ],
+)
+
+cc_binary(
+    name = "verify_token",
```
Comment (JLXIA) [inline src/tools/BUILD]: where this binary is used? Add some comments?

File: src/api_manager/auth/lib/auth_jwt_validator.cc (hunk 4 above)
Comment (qiwzhang) [reply src/api_manager/auth/lib/auth_jwt_validator.cc]: It is Code::UNAUTHENTICATED, please see line 173

File: src/api_manager/auth/lib/auth_jwt_validator_test.cc (hunk 5 above)
Comment (qiwzhang) [reply src/api_manager/auth/lib/auth_jwt_validator_test.cc]: Done

File: src/nginx/t/auth_pkey.t (hunk 1 above)
Comment (qiwzhang) [reply src/nginx/t/auth_pkey.t]: Done

File: src/tools/verify_token.cc (hunk 3 above)
Comment (qiwzhang) [reply src/tools/verify_token.cc]: Done

File: src/tools/verify_token.cc (hunk 2 above)
Comment (qiwzhang) [reply src/tools/verify_token.cc]: Done

File: src/tools/verify_token.cc [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+}  //  namespace
+
+void print_usage() {
+  std::cerr << "Invalid argument.\n"
+               "Usage: verify_token token_file public_key_file\n";
+}
+
+std::string read_file(const char *file) {
+  std::cout << "==Reading file: " << file << std::endl;
+  FILE *fp = fopen(file, "r");
```
Comment (kh-chang) [inline src/tools/verify_token.cc]: should we add nullptr check for fp?

File: src/tools/BUILD (hunk 6 above)
Comment (qiwzhang) [reply src/tools/BUILD]: Done

**Pull request merged**